clang/test/CodeGenSYCL/sycl-device.cpp

```diff
@@ -3,6 +3,78 @@
 
 int bar(int b);
 
+int bar10(int a) { return a + 10; }
+int bar20(int a) { return a + 20; }
+
+class A {
+public:
+  // CHECK-DAG: define linkonce_odr spir_func void @_ZN1A3fooEv
+  // CHECK-DAG: define spir_func i32 @_Z5bar20i
+  __attribute__((sycl_device)) void foo() { bar20(10); }
+
+  // CHECK-DAG: define linkonce_odr spir_func void @_ZN1AC1Ev
+  __attribute__((sycl_device))
+  A() { bar10(10); }
+  // CHECK-DAG: define spir_func i32 @_Z5bar10i
```

Comment:
Please, move this check before constuctor body to make it consistent with foo member function definition check.

Reply:
Sure, done.

Reply:
I don't see the changes. Did you forget to update the branch?

Reply:
No, I didn't. Please update your page or see https://github.com/intel/llvm/pull/1295/commits/1c3036c5dd436377dc07ade6e2277e7242b3df96 .